x-pack/plugins/endpoint/server/routes/whitelist.test.ts

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  IClusterClient,
+  IRouter,
+  IScopedClusterClient,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+  RouteConfig,
+} from 'kibana/server';
+import {
+  elasticsearchServiceMock,
+  httpServerMock,
+  httpServiceMock,
+  loggingServiceMock,
+} from '../../../../../src/core/server/mocks';
+import { WhitelistRule, WhitelistSet } from '../../common/types';
+import { SearchResponse } from 'elasticsearch';
+import { registerWhitelistRoutes } from './whitelist';
+import { EndpointConfigSchema } from '../config';
+import * as data from '../test_data/all_whitelist_data.json';
+
+// A fake elasticsearch error message
+const mockElasticSearchError = 'This is some error that you would expect from elasticsearch';
+
+describe('test whitelist route', () => {
+  let routerMock: jest.Mocked<IRouter>;
+  let mockResponse: jest.Mocked<KibanaResponseFactory>;
+  let mockClusterClient: jest.Mocked<IClusterClient>;
+  let mockScopedClient: jest.Mocked<IScopedClusterClient>;
+  let routeHandler: RequestHandler<any, any, any>;
+  let routeConfig: RouteConfig<any, any, any, any>;
+
+  beforeEach(() => {
+    mockClusterClient = elasticsearchServiceMock.createClusterClient();
+    mockScopedClient = elasticsearchServiceMock.createScopedClusterClient();
+    mockClusterClient.asScoped.mockReturnValue(mockScopedClient);
+    routerMock = httpServiceMock.createRouter();
+    mockResponse = httpServerMock.createResponseFactory();
+    registerWhitelistRoutes(routerMock, {
+      logFactory: loggingServiceMock.create(),
+      config: () => Promise.resolve(EndpointConfigSchema.validate({})),
+    });
+  });
+
+  it('should create a whitelist rule from a POST request', async () => {
+    const expectedComment = 'Hi this is a comment on a whitelist rule';
+    const expectedPath = '/some/path/to/a/file';
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: '/api/endpoint/whitelist',
+      body: {
+        alert_id: '1234',
+        comment: expectedComment,
+        file_path: expectedPath,
+      },
+      method: 'post',
+    });
+
+    const response = { errors: null };
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/whitelist')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
```

Comment:
We should create a `mockContext` variable and reuse it. It seems to be the same for all the tests here.

Reply:
All of these have to be reworked, I'll get to them. Focusing on "integration" style tests for now. I don't really think these unit tests do much in the grand scheme of things.

Reply:
@alexk307 I think we should get rid of these personally, and just use integration tests. Unit tests should be for internal structures.

Reply:
agreed. I'll leave the ones I have now and only add to integ tests.